### ISSUE TITLE
WebTextField setBorder doesn't work after "setLeadingComponent"

### ISSUE THREAD

**Issue by AlpacaMan233** [mgarin/weblaf#416]
After `setLeadingComponent`, WebTextField's'  `setBorder`  doesn't work. Here is my code.

```
public class MainWindow extends WebFrame{
    public MainWindow(){
        setSize(480,800);
        setLayout(new BorderLayout());
        WebTextField searchField =new WebTextField("",20);
//        searchField.setLeadingComponent(new WebImage(new ImageIcon(getClass().getResource("search.png"))));
        searchField.setBorder(BorderFactory.createEmptyBorder(100, 100, 100, 0));
        add(searchField,BorderLayout.NORTH);
        setDefaultCloseOperation(WindowConstants.DISPOSE_ON_CLOSE);
        setVisible(true);
    }
}
```

![1](https://cloud.githubusercontent.com/assets/22738837/19549627/f2bf7892-96d7-11e6-8990-bce0103234a6.png)
![2](https://cloud.githubusercontent.com/assets/22738837/19549630/f710836e-96d7-11e6-9071-b5df91af0ac6.png)

**Comment by mgarin** [comment]
That might be a case as border is used to provide appropriate spacing for the text area in the field and it is calculated automatically based on the leading/trailing components if any are set.

I doubt you want to manually calculate the required border, so the best way is to avoid setting the border directly. I could point you at a better solution if you would say what exactly you tried to achieve with a custom border :)

Also, it is important for me to know - which WebLaF version you are using? - v1.28 release (old version), v1.29 pre-release builds (which can be downloaded from the site) and v1.2.9 snapshot builds are all quite different and have lots of differences in how component borders are handled.

**Comment by AlpacaMan233** [comment]
I'm using v1.2.9 pre-release. I find my app look different from demo. My text field doesn't have appropriate spacing.
![1](https://cloud.githubusercontent.com/assets/22738837/19567821/cf464748-9721-11e6-8620-302fd05a24f8.png)
![qq](https://cloud.githubusercontent.com/assets/22738837/19567836/d96ebec6-9721-11e6-9309-b191751b71a7.png)

**Comment by mgarin** [comment]
By default spacings are minimum as field cannot really predict what kind of icon or font you will use, and those affect the final look a lot.

Also the field on the screenshot (with orange border) uses a completely different painter and settings - it was basically a demo of styling field using android 9-patch images. Spacings are defined on the image patches there in addition to existing padding/margin.

Basically what I would recommend in this case - set larger padding or margin for the field icon component. In your case it is `WebImage` component and since you are using older WebLaF version - you can only provide `margin` setting there, but it should be sufficient. Set it to something like `(4,4,4,4)` and it will extend the field for you :)

**Comment by mgarin** [comment]
Also you might consider trying out newer WebLaF version (currently SNAPSHOT builds of `styling` branch) as it is miles ahead on the amount of improvements, fixes and functionality in general and it is also quite stable. Though it is up to you to decide of course.

**Comment by mgarin** [comment]
@AlpacaMan233 Did you manage to solve this problem or you still need further assistance?

**Comment by mgarin** [comment]
I'll close this issue for now, but feel free to reopen if you still have any related questions or problems.
